engines/python/src/main/java/ai/djl/python/engine/RollingBatch.java

```diff
@@ -342,7 +342,7 @@ void addResponse(byte[] json, Map<String, String> properties) {
                         break;
                 }
             }
-            if ((nextToken == null || nextToken.isEmpty()) && code == null) {
+            if ((nextToken == null || nextToken.isEmpty()) && !last) {
```

Comment:
For tool calling, the last token might be empty as well.

Reply:
the test you put in caught this issue :)